applications/minotari_node/src/main.rs

```diff
@@ -153,16 +159,44 @@ fn main_inner() -> Result<(), ExitError> {
     #[cfg(all(unix, feature = "libtor"))]
     if config.base_node.use_libtor && config.base_node.p2p.transport.is_tor() {
         let tor = Tor::initialize()?;
+        let data_dir = tor.data_dir().clone();
         tor.update_comms_transport(&mut config.base_node.p2p.transport)?;
         tor.run_background();
+
         debug!(
             target: LOG_TARGET,
             "Updated Tor comms transport: {:?}", config.base_node.p2p.transport
         );
-    }
+
+        if std::fs::metadata(libtor_handshake_file_name()).is_ok() {
```

Comment:
Why not just keep the tor data_dir in a variable?

Reply:
That is because we have multiple process exit paths